x-pack/plugins/threat_intelligence/public/modules/indicators/components/indicator_barchart_legend_action/indicator_barchart_legend_action.tsx

```diff
@@ -45,25 +47,27 @@ export const IndicatorBarchartLegendAction: VFC<IndicatorBarchartLegendActionPro
   ];
 
   return (
-    <EuiPopover
-      data-test-subj={POPOVER_BUTTON_TEST_ID}
-      button={
-        <EuiToolTip content={BUTTON_LABEL}>
-          <EuiButtonIcon
-            aria-label={BUTTON_LABEL}
-            iconType="boxesHorizontal"
-            iconSize="s"
-            size="xs"
-            onClick={() => setPopover(!isPopoverOpen)}
-          />
-        </EuiToolTip>
-      }
-      isOpen={isPopoverOpen}
-      closePopover={() => setPopover(false)}
-      panelPaddingSize="none"
-      anchorPosition="downLeft"
-    >
-      <EuiContextMenuPanel size="s" items={popoverItems} />
-    </EuiPopover>
+    <div css={styles.contextMenu}>
+      <EuiPopover
+        data-test-subj={POPOVER_BUTTON_TEST_ID}
+        button={
+          <EuiToolTip content={BUTTON_LABEL}>
+            <EuiButtonIcon
+              aria-label={BUTTON_LABEL}
+              iconType="boxesHorizontal"
+              iconSize="s"
+              size="xs"
+              onClick={() => setPopover(!isPopoverOpen)}
```

Comment:
Hey, in this particular situation, the recommended approach is to use `https://reactjs.org/docs/hooks-reference.html#functional-updates` - check it out :)

Reply:
Also, maybe we could extract this handler with `useCallback`?

Reply:
thanks I wasn't aware of the functional updates, it's cool as it prevent rerendering if the value is identical!

It seems to me that the useCallback here is overkill. We're not really rendering a complex component, just a EuiContextPanel with 3 icons and 3 texts... No?